### ISSUE TITLE
LR_finder per token

### ISSUE THREAD

**Issue by helpmefindaname** [flairNLP/flair#2455]
I am using the current lr finder with a transformer-embedding for ner on a modification flair where I get the following logs:

2021-09-29 12:54:46,651 lr: 3.6307805477010137e-07, loss: 263.8589783114512, best_loss: 645.8231811523438
2021-09-29 12:54:49,414 lr: 4.36515832240166e-07, loss: 272.0130910645663, best_loss: 324.7037353515625
2021-09-29 12:54:49,414 lr: 4.36515832240166e-07, loss: 272.0130910645663, best_loss: 324.7037353515625
2021-09-29 12:54:53,191 lr: 5.248074602497725e-07, loss: 296.9264952525988, best_loss: 479.1085510253906
2021-09-29 12:54:53,191 lr: 5.248074602497725e-07, loss: 296.9264952525988, best_loss: 479.1085510253906
2021-09-29 12:54:57,708 lr: 6.309573444801933e-07, loss: 324.03730629836576, best_loss: 544.8917236328125
2021-09-29 12:54:57,708 lr: 6.309573444801933e-07, loss: 324.03730629836576, best_loss: 544.8917236328125
2021-09-29 12:54:59,748 lr: 7.585775750291837e-07, loss: 309.8983984327399, best_loss: 183.16525268554688
2021-09-29 12:54:59,748 lr: 7.585775750291837e-07, loss: 309.8983984327399, best_loss: 183.16525268554688
2021-09-29 12:55:02,250 lr: 9.120108393559096e-07, loss: 306.78769774358204, best_loss: 183.16525268554688

The loss looked ridiculously large to me, so I investigated and noticed that opposed to the training, the loss doesn't get averaged ov the token count.

I think that this averaging (and also adding the logs to the master) would be beneficial, as otherwise learning rates whose respective batches have short sentences will be preferred, no matter of the actual impact.

**Comment by helpmefindaname** [comment]
@alanakbik I am working on an older commit of the master. I see that you did some changes, which aren't added on my version.
However those changes do not change the behavior of the loss calculation, but changes the way the average per batch is calculated.

I am rather referring to the loss per token in the training: https://github.com/flairNLP/flair/blob/master/flair/trainers/trainer.py#L469 while the lr finder doesn't divide: https://github.com/flairNLP/flair/blob/master/flair/trainers/trainer.py#L926
